lnwallet/script_utils.go

```diff
@@ -106,6 +107,27 @@ func GenFundingPkScript(aPub, bPub []byte, amt int64) ([]byte, *wire.TxOut, erro
 	return witnessScript, wire.NewTxOut(amt, pkScript), nil
 }
 
+// spendP2WPKH generates the witness stack required to redeem a P2WPKH output.
+func spendP2WPKH(signer Signer, signDesc *SignDescriptor,
```

Comment:
Great that we have generic sign descriptors for regular outputs! Can this be consolidated with `CommitNoDelay` for maintainability? `CommitNoDelay` could just return the output of this method

Reply:
[CommitSpendNoDelay](https://github.com/lightningnetwork/lnd/blob/6989316b11c51922b4c6ae3507ac06680ec530b9/lnwallet/script_utils.go#L1035:6) tweaks the public key, so I don't think we'd want that here.